audio/wave_shaper_node.rs

```diff
@@ -116,3 +165,30 @@ impl AudioNodeEngine for WaveShaperNode {
 
     make_message_handler!(WaveShaperNode: handle_waveshaper_message);
 }
+
+impl WaveShaperNode {
+    fn apply_curve(buf: &mut [f32], curve: &Vec<f32>) {
+        buf.iter_mut().for_each(|sample| {
+            let len = curve.len();
+            let curve_index: f32 = ((len - 1) as f32) * (*sample + 1.) / 2.;
```

Comment:
worth  moving `(len - 1) / 2` and `len` out of the closure. Probably will be done by the compiler anyway, but doesn't hurt.

Reply:
agreed, I'll fix it up tomorrow